bot/seasons/evergreen/issues.py

```diff
@@ -3,55 +3,60 @@
 
     def __init__(self, bot: commands.Bot):
         self.bot = bot
 
-    @commands.command(aliases=("issues",))
-    @override_in_channel()
-    async def issue(
-        self, ctx: commands.Context, number: int, repository: str = "seasonalbot", user: str = "python-discord"
-    ) -> None:
-        """Command to retrieve issues from a GitHub repository."""
-        api_url = f"https://api.github.com/repos/{user}/{repository}/issues/{number}"
-        failed_status = {
-            404: f"Issue #{number} doesn't exist in the repository {user}/{repository}.",
-            403: f"Rate limit exceeded. Please wait a while before trying again!"
-        }
-
-        async with self.bot.http_session.get(api_url) as r:
-            json_data = await r.json()
-            response_code = r.status
-
-        if response_code in failed_status:
-            return await ctx.send(failed_status[response_code])
-
-        repo_url = f"https://github.com/{user}/{repository}"
-        issue_embed = discord.Embed(colour=Colours.bright_green)
-        issue_embed.add_field(name="Repository", value=f"[{user}/{repository}]({repo_url})", inline=False)
-        issue_embed.add_field(name="Issue Number", value=f"#{number}", inline=False)
-        issue_embed.add_field(name="Status", value=json_data["state"].title())
-        issue_embed.add_field(name="Link", value=json_data["html_url"], inline=False)
+    @commands.command(aliases=("pr",))
+    async def issues(self, ctx: commands.Context, number: int, repository: str = "seasonalbot",
+                     user: str = "python-discord") -> None:
+        """Command to get issues/pull request from GitHub."""
+        url = f"https://api.github.com/repos/{user}/{repository}/issues/{str(number)}"
+        mergeURL = f"https://api.github.com/repos/{user}/{repository}/pulls/{str(number)}/merge"
 
-        description = json_data["body"]
-        if len(description) > 1024:
-            placeholder = " [...]"
-            description = f"{description[:1024 - len(placeholder)]}{placeholder}"
-
-        issue_embed.add_field(name="Description", value=description, inline=False)
+        async with self.bot.http_session.get(url) as r:
+            json_data = await r.json()
 
-        await ctx.send(embed=issue_embed)
+        if r.status in respValue:
+            return await ctx.send(f"[{str(r.status)}] {respValue.get(r.status)}")
+
+        if "issues" in json_data.get("html_url"):
+            if json_data.get("state") == "open":
+                iconURL = icons.get("issue")
+            else:
+                iconURL = icons.get("issue_closed")
+        else:
+            async with self.bot.http_session.get(mergeURL) as m:
+                if json_data.get("state") == "open":
+                    iconURL = icons.get("pull-request")
+                elif m.status == 204:
+                    iconURL = icons.get("merge")
+                else:
+                    iconURL = icons.get("pull-request-closed")
```

Comment:
This could probably do with a brief comment or two explaining how a pull request is being differentiated from a regular issue, and why a second API query is necessary if a PR is being requested by the command.

Reply:
I don't think this code block has been sufficiently explained. Why is `json_data.["html_url"]` being used as a criteria for whether or not another API query is made? What information is unavailable to the first API query that is necessary to determine the status of the pull request? 

To someone who isn't familiar with the GH API it isn't clear that issues and PRs, which are treated the same by GH's URL patterns, are treated as different things by the current API.

Reply:
This is old code but I remember that the return for it was generic and you had to re-request when  it was a PR. When the output did not contain issues then you knew that you had to call the PR URL which is a different endpoint. There is no other way around this. You need to call it to see wherehter it is PR (open), PR (merged) or PR (closed).

Reply:
If that is a sufficient reason for the second call then I can comment in what is happening in more depth and then commit. Please let me know.

Reply:
That seems fine. The advantage of having this commented in the code is so you don't have to remember that the information returned by the issue endpoint for a PR does not contain the information on whether or not the PR is open, closed, or merged so you have to query the PR endpoint instead.

Reply:
Will get this done in the morning!

Reply:
Done.